packages/client/hmi-client/src/petrinet/petrinet-renderer.ts

```diff
@@ -44,25 +46,107 @@ export class PetrinetRenderer extends graphScaffolder.BasicRenderer<NodeData, Ed
 			(d) => d.data.type === 'T' || d.data.type === NodeType.Transition
 		);
 
+		// transitions
 		transitions
 			.append('rect')
 			.classed('shape', true)
 			.attr('width', (d) => d.width)
 			.attr('height', (d) => d.height)
-			.style('fill', '#89BEFF')
-			.style('stroke', '#888');
+			.attr('y', (d) => -d.height * 0.5)
+			.attr('x', (d) => -d.width * 0.5)
+			.style('fill', '#FFF')
+			.attr('stroke', '#AAA')
+			.attr('stroke-width', 2);
+
+		// transitions drag handles
+		transitions
+			.append('circle')
+			.classed('shape', true)
+			.classed('no-drag', true)
+			.attr('cx', (d) => d.width * 1.2)
+			.attr('r', (d) => d.width * 0.2)
+			.attr('fill', '#FFF')
+			.attr('stroke', '#AAA')
+			.attr('stroke-width', 2)
+			.style('opacity', 0);
 
+		transitions
+			.append('circle')
+			.classed('shape', true)
+			.classed('no-drag', true)
+			.attr('cy', (d) => d.height * 1.2)
+			.attr('r', (d) => d.width * 0.2)
+			.attr('fill', '#FFF')
+			.attr('stroke', '#AAA')
+			.attr('stroke-width', 2)
+			.style('opacity', 0);
+
+		transitions
+			.append('circle')
+			.classed('shape', true)
+			.classed('no-drag', true)
+			.attr('cx', (d) => -d.width * 1.2)
+			.attr('r', (d) => d.width * 0.2)
+			.attr('fill', '#FFF')
+			.attr('stroke', '#AAA')
+			.attr('stroke-width', 2)
```

Comment:
There are quite a bit of repetitions here, I think you can cut them down by something like:
```
transitions.append('circle').classed('blahblah', true).attr('cx', ...).attr('cy', ...)
transitions.append('circle').classed('blahblah', true).attr('cx', ...).attr('cy', ...)
transitions.append('circle').classed('blahblah', true).attr('cx', ...).attr('cy', ...)
...

transition.selectAll('.blahblah').attr('fill', ...). attr('stroke', ...) ....
```

Also how come there are only three handles instead of 4?

Reply:
ok so you would prefer that the handles were all in one line? there are only 3 handles because if there was a handle up top it might overlap with the node name

Reply:
@mj3cheun I meant you create and set the positions of X-number of handles across X-number of lines, but since they all have the same style you can do it in one shot. Maybe a better example:

```
// Create circle and their position
svg.append('circle').attr('cx', 10).attr('cy', 10).attr('r', 10);
svg.append('circle').attr('cx', 50).attr('cy', 50).attr('r', 10);
svg.append('circle').attr('cx', 100).attr('cy', 100).attr('r', 10);

// Set their style
svg.selectAll('circle').attr('fill', '#f80').attr('stroke', '#888').style('cursor', 'pointer');
```